File: ViewDir.py

gid = 0

def nextGid():
    global gid
    g = gid
    gid += 1
    return g

class ViewDir:
    def __init__(self):
        self.name = ''
        self.pid = -1
        self.subs = None

    def getChild(self,text):
        if self.subs == None:
            self.subs = []

        for vv in self.subs:
            if vv.name == text:
                return vv

        vv = ViewDir()
        vv.name = text;
        vv.pid = nextGid()
        self.subs.append(vv)
        return vv

File: objective-c/ffviews/ffviews/ViewDir.m
//
//  ViewDir.m
//  ffviews
//
//  Created by Peter Kollath on 13/11/14.
//  Copyright (c) 2014 Peter Kollath. All rights reserved.
//

#import "ViewDir.h"


int gid = 1;

@implementation ViewDir


-(id)init
{
    self = [super init];
    if (self)
    {
        self.name = @"";
        self.pid = -1;
    }
    return self;
}

-(ViewDir *)getChild:(NSString *)text
{
    if (self.subs == nil)
    {
        self.subs = [NSMutableArray new];
    }
    
    for (ViewDir * vv in self.subs)
    {
        if ([vv.name isEqualToString:text])
            return vv;
    }
    
    ViewDir * vv = [ViewDir new];
    vv.name = text;
    vv.pid = [ViewDir nextGid];
    [self.subs addObject:vv];
    
    return vv;
}

+(int)nextGid
{
    int g = gid;
    gid++;
    return g;
}


@end
